### PR TITLE
fix(axis): dual axis x positioning of bars

### LINKED ISSUE #759: Bars rendered on top of each other when using separate axes
**Describe the bug**
When a two series bar chart is using two separate y axes, the bars are rendered on  top of each  other. When using a single axis, they are rendered next to each other as expected.

**To Reproduce**
Working with a single axis: https://codesandbox.io/s/naughty-sara-okq4f
Not working with two axes:  https://codesandbox.io/s/elated-newton-luvxc

**Expected behaviour**
The bars should still be rendered side by side (it also seems like there is horizontal space reserved for them)

**Screenshots**
Single axis
Screenshot 2020-07-22 at 18 00 26

Two axes - you can see the blue bar hiding the green one
Screenshot 2020-07-22 at 17 59 43

**Version (please complete the following information):**
 - OS: macos
 - Browser: rChrome
 - Elastic Charts: 19.7.0

**Kibana Cross Issues**
https://github.com/elastic/kibana/issues/72889

### PR DESCRIPTION
**Pull request by nickofthyme**
## Summary

fixes #759 

Using a `groupId` with non-stacked bar series would display the bars on top of each other and not respect nor increment the `offsetIndex` correctly.

Screen Recording 2020-07-22 at 01 14 PM

### Checklist

Delete any items that are not applicable to this PR.

- [x] Proper documentation or storybook story was added for features that require explanation or tutorials
- [x] Unit tests were updated or added to match the most common scenarios

### CODE REVIEW

File: src/chart_types/xy_chart/state/utils/utils.ts [hunk 1]
```diff
@@ -336,6 +340,7 @@ export function computeSeriesGeometries(
       chartTheme,
       enableHistogramMode,
     );
+    orderIndex = counts.barSeries > 0 ? orderIndex + counts.barSeries : orderIndex;
```
Comment (nickofthyme) [inline src/chart_types/xy_chart/state/utils/utils.ts:343]: Need to account for `counts.barSeries` or bars will be hidden behind each other for groups.

**Comment by flash1293** [comment]
It's unreal how fast y'all are addressing incoming bug reports. This PR was opened 1,5 hours after I submitted the bug. I would  say I got lucky with this one but the same happened to me a week ago - great work!

**Comment by nickofthyme** [comment]
@flash1293 are you needing this fix in a specific version of kibana or is master ok? I could backport it into `@elastic/charts@19.7.1` if you need.

**Pull request merged**